src/codewhisperer/activation.ts

```diff
@@ -240,67 +243,98 @@ export async function activate(context: ExtContext): Promise<void> {
-                vscode.window
-                    .showWarningMessage(
-                        CodeWhispererConstants.accessTokenMigrationWarningMessage,
-                        CodeWhispererConstants.accessTokenMigrationWarningButtonMessage,
-                        CodeWhispererConstants.accessTokenMigrationDoNotShowAgain
-                    )
-                    .then(async resp => {
-                        if (resp === CodeWhispererConstants.accessTokenMigrationWarningButtonMessage) {
-                            await vscode.commands.executeCommand('aws.codeWhisperer.refresh')
-                            await showSsoSignIn.execute()
-                        } else if (resp === CodeWhispererConstants.accessTokenMigrationDoNotShowAgain) {
-                            await vscode.window.showInformationMessage(
-                                CodeWhispererConstants.accessTokenMigrationDoNotShowAgainInfo,
-                                'OK'
-                            )
-                            await context.extensionContext.globalState.update(
-                                CodeWhispererConstants.accessTokenMigrationDoNotShowAgainKey,
-                                true
-                            )
-                        }
-                    })
-                context.extensionContext.globalState.update(
-                    CodeWhispererConstants.accessTokenMigrationDoNotShowLastShown,
-                    Date.now()
-                )
+
+            if (t <= CodeWhispererConstants.accessTokenCutOffDate) {
+                maybeShowTokenMigrationWarning()
             } else {
-                await globals.context.globalState.update(CodeWhispererConstants.accessToken, undefined)
-                await vscode.commands.executeCommand('aws.codeWhisperer.refreshRootNode')
-                vscode.window
-                    .showErrorMessage(
-                        CodeWhispererConstants.accessTokenMigrationErrorMessage,
-                        CodeWhispererConstants.accessTokenMigrationErrorButtonMessage
-                    )
-                    .then(async resp => {
-                        if (resp === CodeWhispererConstants.accessTokenMigrationErrorButtonMessage) {
-                            await vscode.commands.executeCommand('aws.codeWhisperer.refresh')
-                            await showSsoSignIn.execute()
-                        } else if (resp === CodeWhispererConstants.accessTokenMigrationDoNotShowAgain) {
-                            await context.extensionContext.globalState.update(
-                                CodeWhispererConstants.accessTokenMigrationDoNotShowAgainKey,
-                                true
-                            )
-                        }
-                    })
+                maybeShowTokenMigrationError()
             }
         }
     }
 
+    function maybeShowTokenMigrationWarning() {
```

Comment:
This function and `maybeShowTokenMigrationError` are almost identical. Please avoid copy-pasting code that is clearly amenable to generalization.

If this code will be removed entirely in the near future, then we don't need to worry about it. But, this feature did have some bugs in previous iterations, and verbose, redundant code can be harder to understand....

Reply:
yes this function will be removed after 1/31